### ISSUE TITLE
Improvement: Wanting to change 16khz requirement to 8khz (or Xkhz ideally)

### ISSUE THREAD

**Issue by lfmunoz** [ggml-org/whisper.cpp#1844]
Currently main program requires
RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 16000 Hz

but all my audios are created as 
RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, mono 8000 Hz

Sure I can convert them using `sox` or `ffmpeg` but would be best if it I can just recompile `main` program to use 8khz. Can anyone provide any guidance and what modifications are required? I'm thinking this should be easy am I wrong? 

Looking at examples/common.cpp  I see 
#define COMMON_SAMPLE_RATE 16000

tried changing to 
#define COMMON_SAMPLE_RATE 8000
but then I get random words in output. 

Tried searching for  COMMON_SAMPLE_RATE din't find anything. Kinda of weird. Is the 16khz hardcoded and that define is just a check? or how does it work? Any guidance would be appreciated.

**Comment by bobqianic** [comment]
It's not feasible because the model was trained using data at a 16KHz sampling rate. When you input audio at 8KHz, it doesn't match the training data's distribution, leading to considerable distortion in the output. AI learns by identifying statistically significant information within the provided dataset.

**Comment by lfmunoz** [comment]
Thanks

**Comment by davesoft11** [comment]
> AI learns by identifying statistically significant information within the provided dataset.

Could you explain this in a little more detail? The difference between an 8KHz and 16KHz shouldn't be one that is significant to AI. In fact, it's the kind of noise that AI wants to ignore in favor of spectral content that is significant to the source, not an artifact of the capture mechanism.

Another way to look at it: if the alternative is that we have to convert them from 8KHz -> 16KHz, then short of using some fancy interpolation filter, it's going to be a lossy process. 

Am I missing something?

**Comment by bobqianic** [comment]
> I'm not sure I understand this. You appear to be saying the FFT size might be cut in half by the code. But the sample rate is also getting cut in half. That should make the bin size the same. Am I wrong?

Right, sorry about my previous comment. I was thinking in 16 kHz. I do think it’s possible, but you’ll need to generate the Mel filter on the fly, because it currently only provides an 80×400 filter.
